csharp/ColumnReader.cs

```diff
@@ -67,12 +74,17 @@ public void Dispose()
 
         public LogicalColumnReader LogicalReader(int bufferLength = 4 * 1024)
         {
-            return LogicalColumnReader.Create(this, bufferLength);
+            return LogicalColumnReader.Create(this, bufferLength, elementTypeOverride: null);
         }
 
         public LogicalColumnReader<TElement> LogicalReader<TElement>(int bufferLength = 4 * 1024)
         {
-            return LogicalColumnReader.Create<TElement>(this, bufferLength);
+            return LogicalColumnReader.Create<TElement>(this, bufferLength, elementTypeOverride: null);
+        }
+
+        public LogicalColumnReader<TElement> LogicalReaderOverride<TElement>(int bufferLength = 4 * 1024)
```

Comment:
Would it be possible to avoid the need for the extra "Override" method and `elementTypeOverride` parameter? If the typed `LogicalReader` method receives a type parameter that's different to the default type provided by the LogicalTypeFactory, but is compatible with the `LogicalReadConverterFactory`, then it should be able to handle this, which would simplify the public API a bit.

Reply:
👍 makes sense